woocommerce/woocommerce
# [CYS] Fetch patterns from the WooCommerce PTK source site

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR introduces a new call to the PTK API with the WooCommerce site (https://wooblockpatterns.wpcomstaging.com/) to get the CYS migrated patterns.

Part of https://github.com/woocommerce/woocommerce/issues/45835



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure you have the `WooCommerce beta tester` plugin installed.
2. Head over to `Tools > WCA Test Helper > Features`.
3. Make sure you see the `pattern-toolkit-full-composability` on the list.
4. Enable it.
5. Go to `wp-admin/admin.php?page=wc-settings&tab=advanced&section=woocommerce_com`, **disable** `Allow usage of WooCommerce to be tracked`, and save.
6. On the same page, check the option again and Save.
7. Go to `wp-admin/tools.php?page=action-scheduler`, search for the `fetch_patterns` action, and make sure it was scheduled.
8. Click on `Run` under the action name if it is still pending and wait until is completed.
9. Create a new page or post.
10. Click on `+`, then in `Patterns`.
<img width="350" alt="Screenshot 2024-06-14 at 12 43 02" src="https://github.com/woocommerce/woocommerce/assets/186112/8e02ff43-720b-4ccc-974b-83d3f5c34925">

11. Check you see the following categories with these patterns inside:

| Category | Patterns |
|--------|--------|
| Intro | `Just Arrived Full Hero`, `Hero Product Split`, `Featured Category Cover Image`
| Featured selling | `Banner`, `Featured Category Focus`, `Hero Product Chessboard`, `Hero Product 3 Split`, `Product Gallery`, `Large Image Product Gallery`, `Featured Category Triple`, `Product Collection: Featured Products 5 Columns`, `Product Collection 3 Columns`, `Product Collection 4 Columns`, `Product Collection 5 Columns` |
| About | `Hero Product Chessboard` |
| Reviews | `Testimonials Single`, `Testimonial 3 Columns` |
| Social media | `Social: Follow us on social media` | 




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
CYS - Fetch patterns from the WooCommerce PTK source site.
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
